src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,30 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.exception.DukeException;
+
+/**
+ * Represents a command.
+ * Parent class of all other types of commands.
+ * A <code>Command</code> object corresponds to a command to perform some action on a <code>TaskList</code>.
+ */
+public abstract class Command {
+
+    /**
+     * Performs an action corresponding to the type of command.
+     * @param tasks Instance of <code>TaskList</code> which stores <code>Task</code> objects.
+     * @param ui Instance of <code>Ui</code> which handles user input and outputs.
+     * @param storage Instance of <code>Storage</code> which stores and loads information to and from the hard disk.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+
+    /**
+     * Checks if this object is an <code>ExitCommand</code>.
+     * @return Whether this command is an exit command.
+     */
+    public boolean isExit() {
```

Comment:
Is there a need for it to be a method? 

I would suggest making this just a boolean value.